core/vm/evm.go

```diff
@@ -229,7 +229,7 @@ func (evm *EVM) Call(caller ContractRef, addr common.Address, input []byte, gas
 			// If the account has no code, we can abort here
 			// The depth-check is already done, and precompiles handled above
 			contract := NewContract(caller, AccountRef(addrCopy), value, gas)
-			contract.SetCallCode(&addrCopy, evm.StateDB.GetCodeHash(addrCopy), code)
+			contract.SetCallCode(&addrCopy, evm.StateDB.GetPoseidonCodeHash(addrCopy), code)
```

Comment:
I think here we should set the `KeccakCodeHash` to the contract. Otherwise, there is [someplace](https://github.com/scroll-tech/go-ethereum/blob/b6cc89fabe893bc9fdbd68ad3764432ae1bcd09a/core/vm/contract.go#L110) to compare the `codehash` in the contract with empty keccak hash.